tracer/src/Datadog.Trace/Debugger/DebuggerSettings.cs

```diff
@@ -0,0 +1,131 @@
+// <copyright file="DebuggerSettings.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.Debugger;
+
+internal class DebuggerSettings
+{
+    private const string DefaultProdSite = "datadoghq.com";
+    private const string DefaultStagingSite = "datad0g.com";
+    private const int DefaultMaxDepthToSerialize = 1;
+    private const int DefaultSerializationTimeThreshold = 150;
+    private const int DefaultConfigurationsPollIntervalSeconds = 1;
+    private const int DefaultUploadBatchSize = 100;
+    private const int DefaultDiagnosticsIntervalSeconds = 3600;
+    private const int DefaultUploadFlushIntervalMilliseconds = 0;
+
+    public DebuggerSettings()
+        : this(configurationSource: null)
+    {
+    }
+
+    public DebuggerSettings(IConfigurationSource configurationSource)
+    {
+        ApiKey = configurationSource?.GetString(ConfigurationKeys.ApiKey);
+        RuntimeId = Util.RuntimeId.Get();
+        ServiceName = configurationSource?.GetString(ConfigurationKeys.ServiceName);
+
+        var exporterSettings = new ExporterSettings(configurationSource);
+
+        var agentUri = exporterSettings.AgentUri.ToString().TrimEnd('/');
+        SnapshotsPath = configurationSource?.GetString(ConfigurationKeys.Debugger.SnapshotUrl)?.TrimEnd('/') ?? agentUri;
```

Comment:
We are not supporting UDS/named pipes for now, you are right, it will be problem here.
I've added check, during `debugger` initialize, so debugger will not start at all.
See this commit da54cdab08ea4053bef0496aacd5f59178af4de1
We will fix it in near future.